tensorflow_probability/python/experimental/mcmc/particle_filter.py

```diff
@@ -344,7 +461,8 @@ def particle_filter(observations,
 
   init_seed, loop_seed = samplers.split_seed(seed, salt='particle_filter')
   with tf.name_scope(name or 'particle_filter'):
-    num_observation_steps = ps.size0(tf.nest.flatten(observations)[0])
```

Comment:
I think this should still be getting the size of the 0-th dimension.  I don't believe the shape of the observations is affected by particles_dim.